rust/util/mod.rs

```diff
@@ -34,11 +34,21 @@ macro_rules! enum_getter {
 
 #[macro_export]
 macro_rules! write_joined {
-    ($f:ident, $iterable:expr, $joiner:tt) => {{
+    ($f:ident, $joiner:expr, $($iterable:expr),* $(,)*) => {{
```

Comment:
We could use an explanation of this new implementation too - I believe you mentioned already it allows us to, once again, pass in a varargs of heterogeneous types

Reply:
That's pretty much that. The macro unrolls into consecutive code blocks roughly equivalent to the previous version, except we keep track of whether the current iterator is the first non-empty one, just so we don't start the output with the joiner.

One other change is that the joiner is now allowed to be an expression, which will be handy when we'll have computed joiners.